APE_Black_Formatting.rst

```diff
@@ -0,0 +1,286 @@
+.. _Black: https://github.com/psf/black
+.. _Django DEP-0008: https://github.com/django/deps/blob/main/final/0008-black.rst
+.. _pandas pre-commit: https://pandas.pydata.org/docs/development/contributing_codebase.html#pre-commit
+.. _coding style: https://docs.astropy.org/en/latest/development/codeguide.html#coding-style-conventions
+
+Formatting Code with Black
+===========================
+
+authors: Tom Aldcroft, Erik Tollerud
+
+date-created: 2022 July 13
+
+date-last-revised: 2022 August 25
+
+date-accepted: <replace with accepted date>
+
+type: Standard Track
+
+status: Discussion
+
+
+Abstract
+--------
+
+This APE proposes that the `astropy core package
+<https://github.com/astropy/astropy>`_ adopt Black_ as the default code format.
+
+Detailed description
+--------------------
+
+Foreword
+^^^^^^^^^
+
+This APE presents a rationale that code formatting with Black_ can be beneficial
+both for the ``astropy`` code development process and for the community.
+Hereafter we refer to the astropy core package as ``astropy``, noting that it is
+distinct from the Astropy project as a whole and coordinated and affiliated
+Astropy packages. While coordinated or affiliated packages may choose to adopt
+this APE's conventions in the future, this APE is concerned solely with the core
+package itself.
+
+Code formatting is controversial and the `astropy-dev thread`_ about Black
+generated a spirited discussion but did not result in a clear consensus. This
+strong division of opinions about automated code formatting is common and has
+been documented in other projects, for instance in `Django DEP-0008`_.
+
+We do not expect unanimous agreement on this APE but do hope to demonstrate the
+value of Black for the collective needs of the community and Astropy Project.
+
+.. _astropy-dev thread: https://groups.google.com/g/astropy-dev/c/6cRJCMgaFyM/
+
+Note that ideas and some text for this APE have been adapted from a similar
+enhancement proposal for Django `Django DEP-0008`_.
+
+Rationale
+^^^^^^^^^
+
+**Consistent and readable code formatting** reduces effort in both code review
```

Comment:
consistent and readable code formatting can be achieved without black. However there has been a pushback from the community for year to enforce linting, even though that step could be done along with pre-commit hooks, and without black. Forcing the pre-commit hook to be run would similarly result of fixed code without separate pep8 commits, so it is kind of an unfair accounting to claim that black would fix this issue.